interfaces/builtin/fwupd.go

```diff
@@ -246,6 +256,41 @@ func (iface *fwupdInterface) AppArmorConnectedPlug(spec *apparmor.Specification,
 	}
 	snippet := strings.Replace(fwupdConnectedPlugAppArmor, old, new, -1)
 	spec.AddSnippet(snippet)
+
+	if release.OnClassic {
```

Comment:
These changes are in the wrong place: it's the slot side that needs to mount `/boot` rather than the plug side.  Moving this code over to the `AppArmorPermanentSlot` method should do the trick.

Also, we've used `implicitSystemPermanentSlot(slot)` in place of `release.OnClassic` to distinguish the implicit `fwupd` slot found on classic systems from snaps that provide the `fwupd` slot.  This way a snap with the slot isn't limited to Ubuntu Core systems.